core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -816,28 +801,30 @@ public SearchSourceBuilder scriptField(String name, Script script, boolean ignor
     }
 
     /**
-     * Sets the boost a specific index will receive when the query is executed
+     * Sets the boost a specific index or alias will receive when the query is executed
      * against it.
      *
      * @param index
-     *            The index to apply the boost against
+     *            The index or alias to apply the boost against
      * @param indexBoost
      *            The boost to apply to the index
      */
     public SearchSourceBuilder indexBoost(String index, float indexBoost) {
-        if (this.indexBoost == null) {
-            this.indexBoost = new ObjectFloatHashMap<>();
+        Objects.requireNonNull(index);
+
+        if (this.indexBoosts == null) {
+            this.indexBoosts = new ArrayList<>();
```

Comment:
if the list is already initialized like I suggested above, this if goes away too